clang/test/CodeGenSYCL/sycl_libdevice_cmath.cpp

```diff
@@ -0,0 +1,148 @@
+// SYCL compilation uses libdevice in order to implement platform specific
+// versions of funcs like cosf, logf, etc. In order for the libdevice funcs
+// to be used, we need to make sure that llvm intrinsics such as llvm.cos.f32
+// are not emitted since many backends do not have lowerings for such
+// intrinsics. This allows the driver to link in the libdevice definitions for
+// cosf etc. later in the driver flow.
+//
+// For SPIR-V compilation, libdevice funcs are used since LLVM intrinsics will
+// not be used if IsMathErrnoDefault evaluates to true, which is the case for
+// SPIR-V. Since IsMathErrnoDefault is a driver setting, we must invoke clangxx
+// for spirv compilation to check that LLVM intrinsics are not being emitted.
+
+// RUN: %clangxx %s -fsycl-device-only -emit-llvm -S -o - | FileCheck %s
```

Comment:
Actually I think you can remove this RUN. This PR isn't modifying this behavior right? IIUC the only thing being modified is behavior for NVPTX backend and so following 2 RUNs should suffice IMO